jupyterhub/orm.py

```diff
@@ -222,9 +267,22 @@ def check_routes(self, user_dict, routes=None):
                 # catch filter bug, either in sqlalchemy or my understanding of its behavior
```

Comment:
Hmm... I wouldn't hold up the PR for this but it would be nice to catch why this is true occasionally.